modules/dhcp/providers/server/infoblox.rb

```diff
@@ -0,0 +1,148 @@
+  class Infoblox < Server
+    def initialize(options = {})
+      super options[:server]
+      @username = options[:username]
+      @password = options[:password]
+    end
+
+   def connect
+     ::Infoblox::Connection.new(:username => Proxy::DhcpPlugin.settings.username, :password => Proxy::DhcpPlugin.settings.password, :host => Proxy::DhcpPlugin.settings.dhcp_server)
+   end
+
+   def delRecord subnet, record
+     validate_subnet subnet
+     validate_record record
+     # TODO: Refactor this into the base class
+     raise InvalidRecord, "#{record} is static - unable to delete" unless record.deleteable?
+     # "Deleting"" a record here means just disabling dhcp
+     connection = connect
+     host = ::Infoblox::Host.find(connection, "ipv4addr" => record.ip)
+     unless host.empty?
+       # if not empty, first element is what we want to edit
+       host = host.first
+       # Select correct ipv4addr object from ipv4addrs array
+       hostip = host.ipv4addrs.find { |ip| ip.ipv4addr == record.ip }
+       hostip.configure_for_dhcp = false
+       # Send object
+       host.put
+     end
+     subnet.delete(record)
+   end
+
+    def addRecord options={}
+      logger.debug "Add Record"
+      record = super(options)
+      
+      connection = connect
+      host = ::Infoblox::Host.find(connection, "ipv4addr" => record.ip)
+      # If empty create:
+      if host.empty?
+        logger.debug "Add Record - Create"
+        # Create new host object
+        host = ::Infoblox::Host.new(:connection => connection)
+        host.name = record.name
+        host.add_ipv4addr(record.ip)
+        post = true
+      else
+        logger.debug "Add Record - Exists using first element"
+        # if not empty, first element is what we want to edit
+        host = host.first
+        post = false
+      end
+      options = record.options
+      # Overwrite values without checking
+      # Select correct ipv4addr object from ipv4addrs array
+      hostip = host.ipv4addrs.find { |ip| ip.ipv4addr == record.ip }
+      logger.debug "Add Record - record.name: #{record.name}, hostip.host #{hostip.host}, record.mac #{record.mac}, record.ip #{record.ip}"
+      logger.debug "Add Record - options[:nextServer] #{options[:nextServer]}, options[:filename] #{options[:filename]}, hostip.ipv4addr: #{hostip.ipv4addr} "
+      raise InvalidRecord, "#{record} Hostname mismatch" unless hostip.host == record.name
+      hostip.mac = record.mac
+      hostip.configure_for_dhcp = true
+      hostip.nextserver = options[:nextServer]
+      hostip.use_nextserver = true
+      hostip.bootfile = options[:filename]
+      hostip.use_bootfile = true
+      ## Test if Host Entry has correct IP
+      raise InvalidRecord, "#{record} IP mismatch" unless hostip.ipv4addr == record.ip
+      # Send object
+      post ? host.post : host.put
+      record
+    end
+    
+    def find_record(record)
+      logger.debug "loadRecord"
```

Comment:
you should call ```super``` here, and check if the record has already been loaded. Otherwise you'll get an error at the end when attempting to create a record that already exists.